### PR TITLE
Does not try to run the provisioner when no configuration is set

### PR DESCRIPTION
**Pull request by imobachgs**
Fixes [bsc#1038630](https://bugzilla.suse.com/show_bug.cgi?id=1038630).

### CODE REVIEW

File: src/lib/configuration_management/clients/configuration_management_finish.rb [hunk 1]
```diff
@@ -24,8 +24,9 @@ def initialize
       # @return [TrueClass,FalseClass] True if configurations have been written;
       #                                otherwise it returns false.
       def write
+        return if config.nil?
```
Comment (ancorgs) [inline src/lib/configuration_management/clients/configuration_management_finish.rb]: The Yardoc says `@return [TrueClass,FalseClass]`. Not sure if `nil` is ok.

File: test/lib/clients/finish_client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    end
+
+    context "when not configuration is set" do
+      let(:config) { nil }
+
+      it "does not run the provisioner" do
+        expect(provision_client).to_not receive(:run)
+      end
+
+      it "returns nil" do
```
Comment (ancorgs) [inline test/lib/clients/finish_client_spec.rb]: Already commented in the code.

File: test/lib/clients/finish_client_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      it "runs the configurator" do
+        expect(configurator).to receive(:prepare)
+        client.write
+      end
+
+      it "runs the provisioner" do
+        expect(provision_client).to receive(:run)
+        client.write
+      end
```
Comment (ancorgs) [inline test/lib/clients/finish_client_spec.rb]: I miss a `it "returns true" do` because of symmetry reasons :D

**Comment by ancorgs** [comment]
LGTM

**Comment by imobachgs** [comment]
Thanks!

**Pull request merged**